vault/dendron.rfc.12-schema-improvements.types.md

```diff
@@ -0,0 +1,116 @@
+---
+id: ffe67303-57be-452c-aa73-f7d7e9c4f00f
+title: Types
+desc: ''
+updated: 1620592717103
+created: 1620582238808
+---
+
+## Goals
+
+Creating the possibility to extend the functionality of schemas by introducing a type value on them.
+
+## Proposal
+
+Adding a type value to schemas, this value would be optional and default to the current functionality. 
+The current functionality should be named `pattern` in my eyes and be the default value for this key in the yaml. 
+
+
+### Proposed Types
+
+The following are Types that I would Propose to create as part of dendrons core functionality, while a pod should have the ability to extend the possible type list, creating a type name like `podname:typename` to prevent conflicts with other core or other types. 
+
+
+#### Pattern
+
+This would be the current functionality. The examples show that they are Equal 
+
+##### Example
+
+```yml
+schemas:
+  - id: example
+    title: Example #defaults to the id
+    pattern: e #defaults to the id if not set
+```
+
+```yml
+schemas:
+  - id: example
+    title: Example #defaults to the id
+    pattern: e #defaults to the id if not set
+    type: pattern
+```
+
+#### Segments
+
+This type would be used to merge multiple hierarchy segments into one for usage in Commands or in UI elements. 
+
+##### Example
+
+
+```yml
+schemas:
+  - id: date
+    title: Example #defaults to the id
+    type: segments 
+    pattern: $0.$1.$2 # this value becomes required,  the ${digit} part describes the hierachy lvl in this scheme, but could use named identifiers as well or static pattern parts.
+    elements: # required for segments, order matters here if number references are used.    
+      - id: year # this would match the reference $0 or $year
+      - id: month # this would match the reference $1 or $month
+      - id: day # this would match the reference $2 or $day        
+```
+
+
+
+#### Enum
+
+This type extends the Pattern type with the option to map the matched values of the Pattern to a title that should be used in the UI. 
+
+##### Example
+
+```yml
+schemas:
+   - id: numericMonth
+     type: enum
```

Comment:
instead of a separate type, would it make sense to make `enum` a property of a schema?

Reply:
On one side this would be an option, on the otherside It just got into my mind that this could be used for conditional child hierarchies as well. 

Will update it acordingly.